### ISSUE TITLE
Undefined behaviour when the label is too long

### ISSUE THREAD

**Issue by Luflosi** [linux-apfs/apfsprogs#5]
Executing `mkapfs -L` with a very long label leads to undefined behaviour because the label string is first copied into a fixed sized buffer and then checked for its length. It should be the other way around or the copying needs to be restricted, for example by using strncpy() instead of strcpy().

**Comment by eafer** [comment]
> Executing `mkapfs -L` with a very long label leads to undefined behaviour because the label string is first copied into a fixed sized buffer and then checked for its length. It should be the other way around or the copying needs to be restricted, for example by using strncpy() instead of strcpy().

I'm not sure I follow. The length of the label seems to be checked early on,
as part of complete_parameters().

That said, the mkapfs is a bit of a hack and there are a few segfaults in
the code, triggered by things like large containers. I don't think it
matters for now, it's not a security risk. They will go away as I implement
the missing features.

**Comment by Luflosi** [comment]
You're right. I just saw an error message related to undefined behaviour, looked at the code a bit, saw one place, where the string is copied with strcpy() and another place where the length is checked and assumed that they were in the wrong order.
Sorry about that.
I looked a bit further and I now think that you simply forgot to add an exit(1); after printing the error message "volume label is too long".
